app/javascript/mastodon/features/account_timeline/index.js

```diff
@@ -170,7 +175,7 @@ class AccountTimeline extends ImmutablePureComponent {
           alwaysPrepend
           append={remoteMessage}
           scrollKey='account_timeline'
-          statusIds={(suspended || blockedBy) ? emptyList : statusIds}
+          statusIds={(suspended || blockedBy || hidden) ? emptyList : statusIds}
           featuredStatusIds={featuredStatusIds}
           isLoading={isLoading}
           hasMore={hasMore}
```

Comment:
These changes will usually not be enough because of `hasMore`: if `hasMore` is true, a “Load more” button will be displayed, and not the `emptyMessage`:

![image](https://user-images.githubusercontent.com/384364/167248336-2f7542ab-6fae-42ed-a9cf-7efef71f2129.png)

Because no status is actually passed, clicking `hasMore` will ask for the same page and not a new one, thus never doing anything.